CMakeLists.txt

```diff
@@ -70,6 +70,7 @@ set(PYBIND11_HEADERS
   include/pybind11/stl.h
   include/pybind11/stl_bind.h
   include/pybind11/typeid.h
+  include/pybind11/detail/class_.h
```

Comment:
Yeah, this PR probably shouldn't introduce a one-off change like that. Renamed `detail/class_.h` -> `class_support.h`. I squashed back the change right away to keep the commits clean.